### ISSUE TITLE
Recommends override SOLV_FAVOR choice (even weak deps are disabled)

### ISSUE THREAD

**Issue by ignatenkobrain** [openSUSE/libsolv#205]
```
job: favor libcrypt
...
- libcrypt-2.25-4.fc26.x86_64[not installed]
- libcrypt-nss-2.25-4.fc26.x86_64[not installed]
installing libcrypt-nss-2.25-4.fc26.x86_64
```

Don't know why it pulls nss one instead..
[debugdata.zip](https://github.com/openSUSE/libsolv/files/1016291/debugdata.zip)

**Comment by j-mracek** [comment]
I have similar problem SOLV_FAVOR and SOLV_DISFAVOR. I have a test case
```
repo @System 0 testtags @System.repo.gz
repo home_jmracek_Projects_ci-dnf-stack_dnf-docker-test_repo_upgrade_2 -99.-1000 testtags home_jmracek_Projects_ci-dnf-stack_dnf-docker-test_repo_upgrade_2.repo.gz
repo home_jmracek_Projects_ci-dnf-stack_dnf-docker-test_repo_upgrade_1 -99.-1000 testtags home_jmracek_Projects_ci-dnf-stack_dnf-docker-test_repo_upgrade_1.repo.gz
system x86_64 rpm @System
poolflags implicitobsoleteusescolors
solverflags allowdowngrade allowvendorchange keepexplicitobsoletes bestobeypolicy keeporphans yumobsoletes
job install oneof TestA-1.0.0-1.noarch@home_jmracek_Projects_ci-dnf-stack_dnf-docker-test_repo_upgrade_2 TestA-1.0.0-2.noarch@home_jmracek_Projects_ci-dnf-stack_dnf-docker-test_repo_upgrade_2 TestA-1.0.0-1.noarch@home_jmracek_Projects_ci-dnf-stack_dnf-docker-test_repo_upgrade_1 TestA-1.0.0-2.noarch@home_jmracek_Projects_ci-dnf-stack_dnf-docker-test_repo_upgrade_1 [setevr,setarch]
job multiversion provides kernel
job multiversion provides kernel-PAE
job multiversion provides installonlypkg(kernel)
job multiversion provides installonlypkg(kernel-module)
job multiversion provides installonlypkg(vm)
job disfavor pkg TestA-1.0.0-2.noarch@home_jmracek_Projects_ci-dnf-stack_dnf-docker-test_repo_upgrade_2
job disfavor pkg TestB-1.0.0-2.noarch@home_jmracek_Projects_ci-dnf-stack_dnf-docker-test_repo_upgrade_2
job disfavor pkg TestA-1.0.0-2.noarch@home_jmracek_Projects_ci-dnf-stack_dnf-docker-test_repo_upgrade_1
job disfavor pkg TestB-1.0.0-2.noarch@home_jmracek_Projects_ci-dnf-stack_dnf-docker-test_repo_upgrade_1
result transaction,problems solver.result
```
Result:
```
install TestA-1.0.0-2.noarch@home_jmracek_Projects_ci-dnf-stack_dnf-docker-test_repo_upgrade_2
install TestB-1.0.0-2.noarch@home_jmracek_Projects_ci-dnf-stack_dnf-docker-test_repo_upgrade_2
```


I tried combination with FAVOR, DISFAVOR, job with "oneof" of "name", but without any change in result. I xpext if I disfavor latest package, the solver will install older version of package if it is possible. Probably I use incorrectly FAVOR, DISFAVOR flags, or it has to be use with or without some solverflag. 
Please @mlschroe can you provide a hint what can be wrong? Thanks a lot.

**Comment by ignatenkobrain** [comment]
@j-mracek without repodata it is hard to tell. Although favor/disfavor doesn't work on versions. See #170.

**Comment by ignatenkobrain** [comment]
just FTR: original case here is that glibc *Recommends* `libcrypt-nss` and even weak deps are disabled, it is still taken into account. and anyhow, even weakdeps would be enabled, SOLV_FAVOR should take precedence. We've discussed this with Michael some time ago on IRC and this is some real bug which should be fixed at some point.

**Comment by ignatenkobrain** [comment]
```
repo system 0 empty
repo available 0 testtags <inline>
#>=Pkg: a 1 1 noarch
#>=Req: b
#>=Rec: b1
#>=Pkg: b1 1 1 noarch
#>=Prv: b
#>=Pkg: b2 1 1 noarch
#>=Prv: b

system x86_64 rpm system

poolflags implicitobsoleteusescolors
solverflags allowvendorchange keepexplicitobsoletes bestobeypolicy keeporphans yumobsoletes ignorerecommended

job install pkg a-1-1.noarch@available
job favor name b2
result transaction,problems,alternatives <inline>
```

This is the testcase which can be used to reproduce problem

**Comment by mlschroe** [comment]
Fixed with commit 2645bc64b322460ac77d0d051de63434d4a90a0a
